lib/webkit_inspection_protocol.dart

```diff
@@ -216,23 +222,34 @@ class WipConnection {
   final _closeController = StreamController<WipConnection>.broadcast();
   final _notificationController = StreamController<WipEvent>.broadcast();
 
-  static Future<WipConnection> connect(String url) {
+  /// Connect to the given url and return a [WipConnection].
+  ///
+  /// On errors from this stream, the [onError] handler is called with the error
+  /// object and possibly a stack trace. The [onError] callback must be of type
+  /// `void Function(Object error)` or `void Function(Object error, StackTrace)`.
+  static Future<WipConnection> connect(String url, {Function? onError}) {
     return WebSocket.connect(url).then((socket) {
-      return WipConnection._(url, socket);
+      return WipConnection._(url, socket, onError: onError);
     });
   }
 
-  WipConnection._(this.url, this._ws) {
-    _ws.listen((data) {
-      var json = jsonDecode(data as String) as Map<String, dynamic>;
+  WipConnection._(this.url, this._ws, {Function? onError}) {
+    void onData(dynamic /*String|List<int>*/ data) {
       _onReceive.add(data);
 
+      var json = jsonDecode(data as String) as Map<String, dynamic>;
       if (json.containsKey('id')) {
         _handleResponse(json);
       } else {
         _handleNotification(json);
       }
-    }, onDone: _handleClose);
+    }
+
+    _ws.listen(
+      onData,
+      onError: onError,
```

Comment:
Thanks Devon! BTW Is there a way to add a test for the error handler? I've tried to use this code by dwds and added some throwing code inside the `onData` function, or calling `_ws.AddError()` right after `_ws.listen`, but the `onError` handler  is not invoked (the exception just falls through)

Reply:
Good question; I'll think about this some (either how to test or at least how to manually verify that the error handler is called when there are ws issues).

Reply:
OK, from looking at this, the websocket itself would need to get an error for the `onError` method to be invoked (it's the one that would invoke it).

I've played around with testing this, but am not able to get the upderlying websocket to call the `onError` method (even when killing the chromedriver process that the websocket is talking to).

I suspect however that the websocket _will_ pass any errors to the `onError` method; it's an open question however if the items reported in the flutter issues are going to show up as websocket errors.

Reply:
I see, I had similar experience... I think we should still check this in and see if we get better reports on closed connections in dwds, nothing to loose there:)